### PR TITLE
Fetch the device list a 2nd time to make sure we have the local key

### LINKED ISSUE #305: Wizard doesn't pull local_key for some devices
The hub that I just picked up, [eMylo 5.0 Bluetooth&3.0 ZigBee Gateway 3 In 1 ](https://www.amazon.com/dp/B09JNM1VW6?psc=1&ref=ppx_yo2ov_dt_b_product_details) (v3.4 device, Firmware: Main Module: v1.4.8 ZigBee Module: v2.4.1), has the same issue.  Wizard didn't pick up the local key (not in tuya-raw), but I was able to get it the "app" way as you mention.  I looked at the "result" raw from the `cloud.getdevices()` call and there is no local_key.

### PR DESCRIPTION
**Pull request by uzlonewolf**
Closes #305

### CODE REVIEW

File: tinytuya/wizard.py [hunk 1]
```diff
@@ -213,7 +226,10 @@ def wizard(color=True, retries=None, forcescan=False, nocloud=False):
             print('\n\n' + bold + 'Unable to save raw file' + dim )
 
     # Find out if we should poll all devices
-    answer = input(subbold + '\nPoll local devices? ' + normal + '(Y/n): ')
+    if quicklist:
```
Comment (jasonacox) [inline tinytuya/wizard.py:229]: Nice!

File: tinytuya/Cloud.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
         getfunctions(deviceid)
         getproperties(deviceid)
         getdps(deviceid)
-        sendcommand(deviceid, commands)
+        sendcommand(deviceid, commands [, uri])
```
Comment (jasonacox) [inline tinytuya/Cloud.py:21]: Good catch

File: tinytuya/Cloud.py [hunk 3]
```diff
@@ -364,6 +403,17 @@ def getdevices(self, verbose=False):
...
         else:
             json_data = self._get_all_devices()
+            users = {}
+            # loop through all devices and build a list of user IDs
+            for dev in json_data['result']:
+                if 'uid' in dev:
+                    users[dev['uid']] = True
+            if users:
+                # we have at least 1 user id, so fetch the device list again to make sure we have the local key
+                # this also gets us the gateway_id for child devices
```
Comment (jasonacox) [inline tinytuya/Cloud.py:413]: Love this!

**Comment by jasonacox** [comment]
Nice... I like the merge approach rather than scanning every ID with missing local_key's.

Testing... 

Do we want to try to get any of the draft changes in on this release?  I think this could go out as v1.12.1.

**Comment by uzlonewolf** [comment]
Up to you.  Only reason I left #308 as a draft was because by the time I was done I was too tired to test it as much as I wanted.  #300 is going to require some slight updates after these 2 are merged.

**Comment by jasonacox** [comment]
To keep it clean (and small batch changes), I released this PR as v1.12.1 (see https://pypi.org/project/tinytuya/1.12.1/).

**Pull request merged**